app/src/main/java/ch/ubique/notifyme/app/utils/ReminderHelper.java

```diff
@@ -31,17 +70,51 @@ public static void setReminder(long alarmTime, Context context) {
 		alarmManager.setExactAndAllowWhileIdle(AlarmManager.RTC_WAKEUP, alarmTime, pendingIntent);
 	}
 
-	private static PendingIntent getPendingIntent(Context context) {
+	private static void removeReminder(PendingIntent pendingIntent, Context context) {
+		AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+		alarmManager.cancel(pendingIntent);
+	}
+
+	private static PendingIntent getPendingIntent(Context context, boolean eightHours) {
 		Intent intent = new Intent(context, ReminderHelper.class);
-		intent.putExtra(KEY_REMINDER_INTENT, KEY_REMINDER_INTENT);
-		return PendingIntent.getBroadcast(context, REMINDER_INTENT_ID, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+		if (eightHours) {
+			intent.putExtra(KEY_EIGHT_HOUR_REMINDER_INTENT, KEY_EIGHT_HOUR_REMINDER_INTENT);
+			return PendingIntent.getBroadcast(context, EIGHT_HOUR_REMINDER_INTENT_ID, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+		} else {
+			intent.putExtra(KEY_REMINDER_INTENT, KEY_REMINDER_INTENT);
+			return PendingIntent.getBroadcast(context, REMINDER_INTENT_ID, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+		}
+	}
+
+	private static PendingIntent getAutoCheckOutPendingIntent(Context context) {
+		Intent intent = new Intent(context, ReminderHelper.class);
+		intent.putExtra(KEY_AUTO_CHECKOUT_INTENT, KEY_AUTO_CHECKOUT_INTENT);
```

Comment:
Shouldn't that be handled with an Intent Action?

Reply:
Yes it should :D I have changed this.